pdns/recursordist/syncres.cc

```diff
@@ -2822,6 +2826,10 @@ bool SyncRes::doCacheCheck(const DNSName& qname, const DNSName& authname, bool w
         else {
           LOG(prefix << qname << ": Entire name '" << qname << "' is negatively cached via '" << ne.d_auth << "' for another " << sttl << " seconds" << endl);
         }
+        if (s_addExtendedResolutionDNSErrors) {
+          // XXX Do we want this?
+          context.extendedError = EDNSExtendedError{0, "Result from negative cache"};
```

Comment:
That would make sense to me, at least when the entire name is negatively cached (and it would be nice to convey that in our EDE message).

Reply:
A push a commit making the distinction